WalletWasabi/WabiSabi/Crypto/CredentialIssuer.cs

```diff
@@ -49,14 +49,24 @@ public class CredentialIssuer
 		/// <param name="credentialIssuerSecretKey">The <see cref="CredentialIssuerSecretKey">coordinator's secret key</see> used to issue the credentials.</param>
 		/// <param name="numberOfCredentials">The number of credentials that the protocol handles in each request/response.</param>
 		/// <param name="randomNumberGenerator">The random number generator.</param>
-		public CredentialIssuer(CredentialIssuerSecretKey credentialIssuerSecretKey, int numberOfCredentials, WasabiRandom randomNumberGenerator)
+		public CredentialIssuer(
+			CredentialIssuerSecretKey credentialIssuerSecretKey,
+			int numberOfCredentials,
+			WasabiRandom randomNumberGenerator,
+			ulong maxAmount)
 		{
+			MaxAmount = maxAmount;
+			RangeProofWidth = (int)Math.Log2(MaxAmount);
```

Comment:
this is incorrect since casting to `int` will round down, but the number of bits required is one more.

it would be simpler to just specify the max amount in terms of the bitwidth, see various comments/responses on slack from yesterday

Reply:
actually, even when it is an integer it's not correct, I think `(int)Math.Ceiling(Math.Log2(MaxAmount+1))` bits are needed in order to be able to represent `MaxAmount` in binary